### PR TITLE
PP-8218 Switch PSP to Stripe tasks

### PR DESCRIPTION
**Pull request by sfount**
Namespace Stripe account info pages by credential ID. Update pages to work contextually in the go live and switching PSP flows.
- add support for Stripe steps in the PSP switching page
- add stripe specific content for switching page
- updates task list item to support showing `completed` for items that
shouldn't be retry-able (disabled but completed items)
- add all stripe steps and update tests to work with new structure

Screenshot 2021-07-13 at 10 05 41

Screenshot 2021-07-13 at 10 06 01

### CODE REVIEW

File: app/controllers/stripe-setup/bank-details/post.controller.js [hunk 1]
```diff
@@ -35,20 +37,34 @@ module.exports = async (req, res, next) => {
...
     })
   }
 
   try {
-    const stripeAccount = await connector.getStripeAccount(req.account.gateway_account_id, req.correlationId)
+    const switchingCredential = getSwitchingCredentialIfExists(req.account)
+    let stripeAccountId
 
-    await updateBankAccount(stripeAccount.stripeAccountId, {
+    if (switchingToCredentials) {
```
Comment (stephencdaly) [inline app/controllers/stripe-setup/bank-details/post.controller.js:49]: `switchingToCredentials` is quite similar to `switchingCredential`
maybe renaming to something like `isSwitchingCredentials` would improve readability

File: app/controllers/stripe-setup/bank-details/post.controller.js [hunk 2]
```diff
@@ -35,20 +37,34 @@ module.exports = async (req, res, next) => {
...
       accountNumber: rawAccountNumber,
       sortCode: rawSortCode,
+      switchingToCredentials,
       errors
     })
   }
 
   try {
-    const stripeAccount = await connector.getStripeAccount(req.account.gateway_account_id, req.correlationId)
+    const switchingCredential = getSwitchingCredentialIfExists(req.account)
```
Comment (stephencdaly) [inline app/controllers/stripe-setup/bank-details/post.controller.js:46]: Moving this variable into the scope of the if block might make it clearer when it's relevant. 

Also should we use the variation of this method that throws an exception if the switching credential does not exist - `getSwitchingCredential`, as we would expect it to if on the switching page?

File: app/views/switch-psp/switch-psp.njk [hunk 3]
```diff
@@ -102,6 +102,11 @@
           formatAccountPathsFor(routes.account.switchPSP.stripeSetup.companyNumber, currentGatewayAccount.external_id, targetCredential.external_id),
           taskList.ENTER_COMPANY_NUMBER
         ) }}
+        {{ taskListItem(
+          "Make a live payment to test your Worldpay PSP",
```
Comment (stephencdaly) [inline app/views/switch-psp/switch-psp.njk:106]: "Stripe"

**Review by stephencdaly — CHANGES_REQUESTED** [review]
Looks good, just a copy-paste error and some optional minor suggestions to improve readability.

File: app/views/switch-psp/switch-psp.njk (hunk 3 above)
Comment (sfount) [reply app/views/switch-psp/switch-psp.njk:106]: 🤦

File: app/controllers/stripe-setup/bank-details/post.controller.js (hunk 2 above)
Comment (sfount) [reply app/controllers/stripe-setup/bank-details/post.controller.js:46]: Really good catch -- I hadn't picked up on that but using the non wrapped method will make any edge cases much clearer

**Review by stephencdaly — APPROVED** [review]
👍

**Pull request merged**